python/test/unit/hopper/test_experimental_tma.py

```diff
@@ -538,3 +540,95 @@ def alloc_fn(size: int, align: int, stream: Optional[int]):
     )
     torch.testing.assert_close(ref_out, A)
     assert "tensormap.cp_fenceproxy.global.shared::cta.tensormap::generic.release.gpu.sync.aligned" in kernel.asm["ptx"]
+
+
+@requires_tma
+@pytest.mark.parametrize("inner_size", [16, 64])
+def test_experimetal_descriptor_load_4d(inner_size):
+    device = "cuda"
+
+    @triton.jit
+    def kernel(Z, desc, inner_size: tl.constexpr):
+        off0 = tl.arange(0, 2)
+        off1 = tl.arange(0, 2)
+        off2 = tl.arange(0, 32)
+        off3 = tl.arange(0, inner_size)
+        x = tl._experimental_descriptor_load(desc, [2, 2, 0, 0], [2, 2, 32, inner_size], tl.dtype("uint8"))
+        out_ptrs = (Z + 2 * 32 * inner_size * off0[:, None, None, None] + 32 * inner_size * off1[None, :, None, None] +
+                    inner_size * off2[None, None, :, None] + off3[None, None, None, :])
+        tl.store(out_ptrs, x)
+
+    x = torch.randint(size=(4, 8, 32, inner_size), low=0, high=100, dtype=torch.uint8).to(device)
+    z_tri = torch.zeros(size=(2, 2, 32, inner_size), dtype=torch.uint8, device=device)
+    desc = TmaDescKernelParam(x.data_ptr(), x.shape, z_tri.shape, 1)
+
+    kernel[(1, )](z_tri, desc, inner_size)
+
+    assert torch.equal(x[2:4, 2:4, :, :], z_tri)
+
+
+@requires_tma
+def test_3d_tma_2d_load():
+    B, M, N, K, BLOCK_M, BLOCK_N = 8, 64, 64, 64, 32, 32
+
+    @triton.jit
+    def kernel(
+        q_desc,
+        k_desc,
+        o_ptr,
+        stride_ob,
+        stride_om,
+        stride_on,
+        BLOCK_M: tl.constexpr,
+        BLOCK_N: tl.constexpr,
+        BLOCK_K: tl.constexpr,
+    ):
+        batch_id = tl.program_id(0)
+        startm = tl.program_id(1) * BLOCK_M
+        startn = tl.program_id(2) * BLOCK_N
+        offs_m = startm + tl.arange(0, BLOCK_M)
+        offs_n = startn + tl.arange(0, BLOCK_N)
+        q = tl._experimental_descriptor_load(q_desc, [batch_id, startm, 0], [1, BLOCK_M, BLOCK_K], tl.float16)
+        k = tl._experimental_descriptor_load(k_desc, [batch_id, 0, startn], [1, BLOCK_K, BLOCK_N], tl.float16)
+        qk = tl.dot(q.reshape(BLOCK_M, BLOCK_K), k.reshape(BLOCK_K, BLOCK_N), out_dtype=tl.float32)
+        o_ptrs = o_ptr + batch_id * stride_ob + offs_m[:, None] * stride_om + offs_n[None, :] * stride_on
```

Comment:
This work but this would definitely not generate efficient code right? My point was that this prevents us from writing an efficient batch matmul